streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -875,7 +878,7 @@ public void run() {
             streamsUncaughtExceptionHandler.accept(e, false);
             // Note: the above call currently rethrows the exception, so nothing below this line will be executed
         } finally {
-            completeShutdown(cleanRun);
+            completeShutdown(cleanRun, true);
```

Comment:
If we close the `StreamThread`,  we should ensure consumers leave their group.
Does it make sense?

Reply:
ah sorry i totally missed this question on my first pass 🤦‍♀️ No, we actually do want to stay in the group and retain the existing default behavior for Streams

if you're curious why the default behavior for Streams is the opposite of the plain consumer client, it's because Streams is stateful which means rebalancing and moving partitions can be a much heavier operation and involve restoring many records from the changelog topics when partitions are shuffled around. By skipping the leave group, we can often avoid an unnecessary extra rebalance

Reply:
Thanks for the explanation!